Content.Client/Fluids/PuddleVisualsSystem.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System.Linq;
+using Content.Shared.Fluids;
+using JetBrains.Annotations;
+using Robust.Client.GameObjects;
+using Robust.Shared.Random;
+
+namespace Content.Client.Fluids
+{
+    [UsedImplicitly]
+    public sealed class PuddleVisualsSystem : VisualizerSystem<PuddleVisualsComponent>
+    {
+        [Dependency] private readonly IRobustRandom _random = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<PuddleVisualsComponent, ComponentInit>(OnComponentInit);
+        }
+
+        private void OnComponentInit(EntityUid uid, PuddleVisualsComponent puddleVisuals, ComponentInit args)
+        {
+            if (!TryComp(uid, out AppearanceComponent? appearance))
+            {
+                Logger.Warning($"Missing AppearanceComponent for PuddleVisualsSystem on entityUid = {uid}");
+                return;
+            }
+
+            if (!TryComp(uid, out SpriteComponent? sprite))
+            {
+                Logger.Warning($"Missing SpriteComponent for PuddleVisualsSystem on entityUid = {uid}");
+                return;
+            }
+
+            if (!appearance.TryGetData(PuddleVisuals.VisualSeed, out float visualSeed))
+            {
+                Logger.Warning($"Missing VisualSeed for PuddleVisualsSystem on entityUid = {uid}");
+                return;
+            }
+
+            var maxStates = sprite.BaseRSI?.ToArray();
+
+            if (maxStates is not { Length: > 0 }) return;
+
+            int intVisualSeed = ((int) (visualSeed * 1000000)); //uses the float seed to generate an arbitrarily large int seed. It is retyped to int for use in the Modulo function.
+
+            int selectedState = intVisualSeed % maxStates.Length; // uses the visualSeed to randomly select an index for which RSI state to use. Modulo is used to get the remainder, so the value will always be between 0 and maxStates.Length.
```

Comment:
uh you should probably just multiply `visualSeed` by `maxStates.Length` then floor/round it